support/builder/s3migrate.sh

```diff
@@ -0,0 +1,360 @@
+    echo "Please enter a target bucket name and press [ENTER]:"
+    read bucket_name
+    # Check if bucket exists, if so create it, if not continue using existing bucket
+    if [ "${s3type}" == 'minio' ]; then
+        if aws --endpoint-url "${AWS_REGION}" s3 ls | grep "${bucket_name}" >/dev/null; then
+            echo "Bucket: ${bucket_name} found!"
+            echo "Using specified bucket."
+            export S3_BUCKET=${bucket_name}
+        else
+            echo "Bucket: ${bucket_name} not found!"
+            echo "Creating bucket: ${bucket_name}"
+            aws --endpoint-url "${AWS_REGION}" s3 mb s3://"${bucket_name}"
+            export S3_BUCKET=${bucket_name}
+        fi
+    else
+        if aws s3api list-buckets | grep "${bucket_name}" >/dev/null; then
+            echo "Bucket: ${bucket_name} found!"
+            echo "Using specified bucket."
+            export S3_BUCKET=${bucket_name}
+        else
+            echo "Bucket: ${bucket_name} not found!"
+            echo "Creating bucket: ${bucket_name}"
+            aws s3api create-bucket --bucket "${bucket_name}" --region "${AWS_REGION}"
+            export S3_BUCKET=${bucket_name}
+        fi
+    fi
+}
+
+setRegion() {
+    # Sets the region used by the bucket
+    if [ "${s3type}" == 'minio' ]; then
+        sgroups=("default" "dev" "prod")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" > /dev/null; then
+                minioIP=$(curl -s localhost:9631/services/builder-minio/"${i}" | jq .sys.ip)
+                echo "We've detected your minio instance at: ${minioIP}!"
+                read -r -p "Would you like to use this minio instance? [y/N] " response
+                if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Setting endpoint to ${minioIP//\"}:9000"
+                    if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.use_ssl | grep "true"> /dev/null; then
+                        export AWS_REGION="https://${minioIP//\"}:9000"
+                    else
+                        export AWS_REGION="http://${minioIP//\"}:9000"
+                    fi
+                    return
+                else
+                    echo ""
+                    echo "==========================================================="
+                    echo "Please enter the minio endpoint URI and press [ENTER]:"
+                    echo "(http://localhost:9000 || https://10.1.250.4:9000)"
+                    echo "==========================================================="
+                    read region_name
+                    export AWS_REGION=${region_name}
+                    return
+                fi
+            fi
+        done
+    else
+        echo ""
+        echo "==========================================================="
+        echo "Please enter the region or the minio endpoint for your bucket and press [ENTER]:"
```

Comment:
Should we be asking about minio endpoints if the s3type is not minio?

Reply:
Nope, excellent catch!